src/omnicore/omnicore.cpp

```diff
@@ -4119,6 +4114,11 @@ int mastercore_handler_block_end(int nBlockNow, CBlockIndex const * pBlockIndex,
         const std::string& msg = strprintf("Shutting down due to failed checkpoint for block %d (hash %s)\n", nBlockNow, pBlockIndex->GetBlockHash().GetHex());
         PrintToLog(msg);
         if (!GetBoolArg("-overrideforcedshutdown", false)) AbortNode(msg, msg);
+    } else {
```

Comment:
I looked at making the change to remove the ```else```.

The shutdown process is threaded and activated by setting flag ```fRequestShutdown``` which is detected by ```DetectShutdownThread()```.

```AbortNode()``` returns (false) after setting the flag, so wouldn't our code continue running until the shutdown detect thread has picked up the flag and interrupted the main threads?

If that's accurate then without the ```else``` the bad state would be written to persistence even without ```overrideforcedshutdown``` which would nullify the fix.

Thoughts?